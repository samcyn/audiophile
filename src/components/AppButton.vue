<script setup lang="ts">
import {
  defineProps,
  withDefaults,
  computed,
  ButtonHTMLAttributes,
  useAttrs,
} from 'vue';

// define components models
interface AppButtonProps extends ButtonHTMLAttributes {
  variant?: 'contained' | 'text' | 'outlined';
}

// set defaults props
const props = withDefaults(defineProps<AppButtonProps>(), {
  variant: 'contained',
});

// pick attrs
const attrs = useAttrs();

// compute classes base on variant type
const classComputed = computed<string>(() => {
  const variant = props.variant;
  let className = 'rounded-none transition ';
  if (variant === 'contained') {
    className += 'bg-orange-100 text-white hover:bg-orange-90';
  }
  if (variant === 'outlined') {
    className +=
      'bg-transparent hover:bg-black-100 border border-black-100 text-black-100 hover:text-white';
  }
  if (variant === 'text') {
    className +=
      'border-0 bg-none text-black-100 opacity-50 hover:text-orange-100 hover:opacity-100';
  }
  return `${className} ${attrs.class}`;
});
</script>

<template>
  <button class="button" :class="classComputed">
    <slot></slot>
  </button>
</template>

<style scoped>
.button {
  font-style: normal;
  font-weight: 700;
  font-size: 13px;
  line-height: 18px;
  letter-spacing: 1px;
  text-transform: uppercase;
  padding: 15px 29.5px 15px 31.5px;
  border-radius: 0;
  outline: 0;
  cursor: pointer;
}
</style>
