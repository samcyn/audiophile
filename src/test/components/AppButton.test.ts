import { mount } from '@vue/test-utils';
import AppButton from '../../components/AppButton.vue';

test('mount AppButton component', async () => {
  expect(AppButton).toBeTruthy();

  const wrapper = mount(AppButton, {
    props: {},
  });

  expect(wrapper.html()).toMatchSnapshot();
});
