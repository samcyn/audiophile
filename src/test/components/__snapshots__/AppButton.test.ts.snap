// Vitest Snapshot v1

exports[`mount AppButton component 1`] = `"<button data-v-b60abf6a=\\"\\" class=\\"button rounded-none transition inline-flex items-center bg-orange-100 text-white hover:bg-orange-90\\" type=\\"button\\"></button>"`;
